gnomad_qc/v4/assessment/generate_ld_data.py

```diff
@@ -0,0 +1,660 @@
+            mt = mt.sample_cols(0.1)
+
+        return mt
+
+    def _generate_ld_mt(
+        version: str = version,
+        is_hgdp: bool = False,
+        test: bool = False,
+        ld_mt_path: str = ld_mt_path,
+        pop: str = "eas",
+        overwrite=overwrite,
+    ) -> hl.MatrixTable:
+
+        if is_hgdp:
+            logger.info("Reading in HGDP_TGP Subset...")
+            mt = hgdp_tgp_subset(dense=True, public=True).mt()
+            mt = mt.annotate_globals(
+                freq_meta=mt.gnomad_freq_meta, freq_index_dict=mt.gnomad_freq_index_dict
+            )
+            mt = mt.annotate_rows(freq=mt.gnomad_freq)
+            mt = mt.annotate_cols(
+                meta=hl.struct(
+                    population_inference=hl.struct(
+                        pop=mt.gnomad_population_inference.pop
+                    )
+                )
+            )
+
+            mt = mt.filter_cols(mt.meta.population_inference.pop == "pop")
+
+            if test:
+                mt = _ld_test_mt(mt, filter_contig=True, sample_cols=True)
+
+            mt = mt.checkpoint(
+                ld_mt_path,
+                overwrite=overwrite,
+            )
+
+        else:
+            logger.info("Reading in gnomAD release HT and VDS...")
+            filter_chrom = None
+            if test:
+                filter_chrom = "chr22"
+            elif chrom:
+                filter_chrom = chrom
+
+            ht_release = hl.read_table(release.release_ht_path(data_type="genomes"))
+            ht_filter = _filter_ht_for_ld(ht_release,label='gen_anc',pop=pop)
+            if filter_chrom:
+                ht_filter = ht_filter.filter(ht_filter.locus.contig==filter_chrom)
+
+            ht_filter = ht_filter.select('freq','filters').checkpoint(new_temp_file())
+
+            vds = basics.get_gnomad_v4_genomes_vds(
+                test=test,
+                annotate_meta=True,
+                release_only=True,
+                split=True,
+                chrom=filter_chrom,
+                filter_variant_ht=ht_filter,
+                entries_to_keep=["GT"],
+            )
+
+            if test or chrom:
+                logger.info(
+                    f"Due to Test or Chrom behavior, only working on {filter_chrom} - do naive coalesce to 1,000 partitions"
+                )
+                mt = vds.variant_data.naive_coalesce(1000)
+            else:
+                mt = vds.variant_data.naive_coalesce(9800)
+
+            logger.info(
+                f"Filtering to relevant cols - only meta - and no VDS/MT rows..."
+            )
+            mt = mt.select_cols(mt.meta)
+            mt = mt.select_rows()
+            logger.info(f"Filtering to {args.pop}...")
+            mt = mt.filter_cols(mt.meta.population_inference.pop == pop)
+
+            # NOTE: QUESTION IF THIS SHOULD BE RAN OR NOT
+            logger.info(f"Checkpointing {pop} on {chrom if chrom else default_str}")
+            mt = mt.checkpoint(new_temp_file())
```

Comment:
Agree that this checkpoint is good since the MT is filtered, and this will improve the LD computation